src/main/java/TaskList.java

```diff
@@ -97,17 +108,26 @@ public void addDeadline(String taskKey) {
             System.out.println(err.getMessage());
             return;
         }
-        Task newTask = new DeadlineTask(taskTitle, deadline);
-        this.itemList.add(newTask);
-        System.out.println(
-                "----------------------------" +
-                        "----------------------------\n" +
-                        "Got it. I've added this task:\n"
-                        + "  " + newTask + "\n"
-                        + "Now you have " + this.itemList.size() + " tasks in the list."
-                        + "\n"
-                        + "--------------------------------------------------------"
-        );
+
+        try {
+            //Accept string
+            //Parse string to datetime, use formatter here to change reading style
+            //when printing, check date time and print format.
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/mm/yyyy HHmm");
+            Task newTask = new DeadlineTask(taskTitle, LocalDateTime.parse(deadline, formatter));
+            this.itemList.add(newTask);
+            System.out.println(
+                    "----------------------------" +
```

Comment:
Great structuring and clear variable names thus far. In this case, I think the hyphenated values can be further abstracted into a String constant to reduce repetition and reduce the length of the print statement. Otherwise, this is well-done!